src/libraries/System.Net.Security/src/System/Net/Security/NegotiateAuthentication.cs

```diff
@@ -0,0 +1,326 @@
+        public bool IsMutuallyAuthenticated => _ntAuthentication?.IsMutualAuthFlag ?? false;
+
+        /// <summary>
+        /// Indicates whether the local side of the authentication is representing
+        /// the server.
+        /// </summary>
+        public bool IsServer => _isServer;
+
+        /// <summary>
+        /// Name of the negotiated authentication package.
+        /// </summary>
+        /// <remarks>
+        /// The negotiated authentication package is only available when the session
+        /// authentication was finished (see <see cref="IsAuthenticated" />). For
+        /// unfinished authentication sessions the value is undefined and usually
+        /// returns the initial authentication package name specified in
+        /// <see cref="NegotiateAuthenticationClientOptions.Package" /> or
+        /// <see cref="NegotiateAuthenticationServerOptions.Package" />.
+        ///
+        /// If the Negotiate package was used for authentication the value of this
+        /// property will be Kerberos, NTLM, or any other specific protocol that was
+        /// negotiated between both sides of the authentication.
+        /// </remarks>
+        public string Package => _ntAuthentication?.ProtocolName ?? _requestedPackage;
+
+        /// <summary>
+        /// Gets target name (service principal name) of the server.
+        /// </summary>
+        /// <remarks>
+        /// For server-side of the authentication the property returns the target name
+        /// specified by the client after successful authentication (see <see cref="IsAuthenticated" />).
+        ///
+        /// For client-side of the authentication the property returns the target name
+        /// specified in <see cref="NegotiateAuthenticationClientOptions.TargetName" />.
+        /// </remarks>
+        public string? TargetName => IsServer ? _ntAuthentication?.ClientSpecifiedSpn : _ntAuthentication?.Spn;
+
+        /// <summary>
+        /// Gets information about the identity of the remote party.
+        /// </summary>
+        /// <returns>
+        /// An <see cref="IIdentity" /> object that describes the identity of the remote endpoint.
+        /// </returns>
+        /// <exception cref="InvalidOperationException">Authentication failed or has not occurred.</exception>
+        /// <exception cref="Win32Exception">System error occurred when trying to retrieve the identity.</exception>
+        public IIdentity RemoteIdentity
+        {
+            get
+            {
+                IIdentity? identity = _remoteIdentity;
+                if (identity is null)
+                {
+                    if (!IsAuthenticated || _ntAuthentication == null)
+                    {
+                        throw new InvalidOperationException(SR.net_auth_noauth);
+                    }
+
+                    if (IsServer)
+                    {
+                        Debug.Assert(!OperatingSystem.IsTvOS(), "Server authentication is not supported on tvOS");
```

Comment:
Should this be Android as well? e.g. managed implementation vs GSSAI?

Reply:
Probably. I'll recheck my local builds to make sure that proper implementation is linked on Android.

Reply:
I remember now. `NegotiateStream` (and many related classes) are marked `[UnsupportedOSPlatform("tvos")]`. They don't have the same attribute on Android (which is separate issue). This basically allowed the compilation to proceed even though its unreachable branch that goes into `PlatformNotSupportedException`.

I plan to follow up with more PRs that enable client-side `NegotiateStream` on tvOS, eventually, so I am not sure it's worth updating the assert and restarting the builds. Of course I can do it if you think it's worth it.